pkg/daemon/update.go

```diff
@@ -93,9 +99,56 @@ func getNodeRef(node *corev1.Node) *corev1.ObjectReference {
 	}
 }
 
-// finalizeAndReboot is the last step in an update(), and it can also
-// be called as a special case for the "bootstrap pivot".
-func (dn *Daemon) finalizeAndReboot(newConfig *mcfgv1.MachineConfig) (retErr error) {
+func reloadCrioConfig() error {
+	_, err := runGetOut("pkill", "-HUP", "crio")
```

Comment:
One of the massive benefits of systemd over baseline Unix is that by using cgroups, it applies a rigorous *structure* to the prior total Wild West of Unix processes you could identify by grepping their name or maybe racy PID files.

In particular, this invocation will send SIGHUP to e.g. a process running in a regular user pod if it happens to be named "crio".

What you want instead is `systemctl reload crio` - systemd is actively tracking the PID that it launched and can send the signal in a race-free way to exactly the correct process.

Reply:
Ah for some reason I thought crio didn't have it defined. I can confirm that it indeed should be doing the same thing:

```
# systemctl cat crio.service
...
ExecReload=/bin/kill -s HUP $MAINPID
...
```

And running a reload:
```
Nov 30 23:23:17 ip-10-0-131-15 systemd[1]: Reloading Open Container Initiative Daemon.
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.304012464Z" level=info msg="Reloading configuration"
Nov 30 23:23:17 ip-10-0-131-15 systemd[1]: Reloaded Open Container Initiative Daemon.
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.309633738Z" level=info msg="Updating config from file /etc/crio/crio.conf.d/0-default"
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.309845337Z" level=info msg="Updating config from path /etc/crio/crio.conf.d"
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.310093463Z" level=info msg="Applied new registry configuration: &{Registries:] UnqualifiedSearchRegistries:[registry.access.redhat.com docker.io]}"
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.310131904Z" level=info msg="No seccomp profile specified, using the internal efault"
Nov 30 23:23:17 ip-10-0-131-15 crio[1507]: time="2020-11-30 23:23:17.310144510Z" level=info msg="Set config seccomp_profile to \"\""
```

Seems like its doing what we want. I'll defer to @sinnykumari since she is more knowledgeable on whether that is correct.

Reply:
Ah, didn't know `crio reload` already does the same thing, definitely nicer way to go.
Useful learning and something I will keep in mind :)